From 6266e691e0cc52570098309a8cd20ef3e772f35e Mon Sep 17 00:00:00 2001
From: Vishal Raval <vishal.raval@einfochips.com>
Date: Mon, 13 Mar 2023 14:26:34 +0000
Subject: [PATCH] Normalize_pointcloud

---
 corelib/src/util3d.cpp | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/corelib/src/util3d.cpp b/corelib/src/util3d.cpp
index 8421f078..bdb6bfdd 100644
--- a/corelib/src/util3d.cpp
+++ b/corelib/src/util3d.cpp
@@ -231,9 +231,19 @@ pcl::PointXYZ projectDepthTo3D(
 		cy = cy > 0.0f ? cy : float(depthImage.rows/2) - 0.5f; //cameraInfo.K.at(5)
 
 		// Fill in XYZ
-		pt.x = (x - cx) * depth / fx;
-		pt.y = (y - cy) * depth / fy;
-		pt.z = depth;
+		//pt.x = (x - cx) * depth / fx;
+		//pt.y = (y - cy) * depth / fy;
+		//pt.z = depth;
+		
+		//Added to Normalize the depth image
+		float xz = (x - cx) / fx;
+		float yz = (y - cy) / fy;
+		float z= depth / sqrt(1 + xz*xz + yz*yz);
+		pt.x = xz * z;
+		pt.y = yz * z;
+		pt.z = z;
+
+
 	}
 	else
 	{
-- 
2.25.1

